### ISSUE TITLE
nvme disconnect for rdma transport will last a considerable amount of time

### ISSUE THREAD

**Issue by liuqinfei** [spdk/spdk#2742]
# Sighting report

When i run "nvme disconnect -d /dev/nvme0n1" for rdma transport. It will last more than 10 seconds untill this step finish.

## Expected Behavior

Just like TCP transport. The disconnect step should finish immediately.

## Current Behavior

Now, it last more than 10 seconds. This is unacceptable for scenarios where a large number of nvmes are connected.

## Possible Solution

Not yet.

## Steps to Reproduce

1. nvme connect -t rdma --traddr 192.168.0.86 -s 6001 -n nqn.2016-06.io.spdk:cnode1
2. nvme disconnect -d /dev/nvme2

![image](https://user-images.githubusercontent.com/42903181/197477102-3c6369d6-a537-4c6f-babd-ecf40140a321.png)
## Context (Environment including OS version, SPDK version, etc.)

NA

**Comment by Comphix** [comment]
There is no other operation like IO between the connect and disconnect operation?

Also any dmesg logs after running the disconnect command? And any logs from NVMe-oF Target side?

If there is no IO operation, not sure whether this is with the Keep Alive command. You may have a try to disable or set a small value for Keep Alive. The related RPC is bdev_nvme_set_options and related parameter is keep_alive_timeout_ms.

```
nvme connect -t rdma --traddr 192.168.0.86 -s 6001 -n nqn.2016-06.io.spdk:cnode1
nvme disconnect -d /dev/nvme2
```

**Comment by AlekseyMarchuk** [comment]
Hi,

What is your OS version and NIC model and SPDK version? Have you tried to use SPDK initiator (nvme_perf or bdevperf) to check if the problem still exists? I'm 99% sure that this problem is caused by kernel modules on the initiator side

**Comment by wanghailiangx** [comment]
I can not reproduce, the following are my detailed steps,
```
./build/bin/nvmf_tgt &
./scripts/rpc.py bdev_nvme_attach_controller -b "Nvme0" -t "pcie" -a 0000:5f:00.0
./scripts/rpc.py nvmf_create_transport -t rdma -u 8192
./scripts/rpc.py nvmf_create_subsystem nqn.2016-06.io.spdk:cnode1 -a -s SPDK1
./scripts/rpc.py nvmf_subsystem_add_ns nqn.2016-06.io.spdk:cnode1 Nvme0n1
./scripts/rpc.py nvmf_subsystem_add_listener nqn.2016-06.io.spdk:cnode1 -t tcp -a 10.0.0.2 -s 6001
nvme connect -t rdma --traddr 10.0.0.2 -s 6001 -n nqn.2016-06.io.spdk:cnode1
nvme disconnect -d /dev/nvme0  (about 1s)
```
kernel 5.4.7

**Comment by liuqinfei** [comment]
> There is no other operation like IO between the connect and disconnect operation?
> 
> Also any dmesg logs after running the disconnect command? And any logs from NVMe-oF Target side?
> 
> If there is no IO operation, not sure whether this is with the Keep Alive command. You may have a try to disable or set a small value for Keep Alive. The related RPC is bdev_nvme_set_options and related parameter is keep_alive_timeout_ms.
> 
> ```
> nvme connect -t rdma --traddr 192.168.0.86 -s 6001 -n nqn.2016-06.io.spdk:cnode1
> nvme disconnect -d /dev/nvme2
> ```

Yes, i do fio test between the connect and disconnect operation. But i do nvme disconnect after i finish the fio test.
I did not see any abnormal log on the NVMe-oF target side.

**Comment by liuqinfei** [comment]
> t is your OS version and NIC model and SPDK version? Have you tried to use SPDK initiator (nvme_perf or bdevperf) to che



> Hi,
> 
> What is your OS version and NIC model and SPDK version? Have you tried to use SPDK initiator (nvme_perf or bdevperf) to check if the problem still exists? I'm 99% sure that this problem is caused by kernel modules on the initiator side

The NVMe-oF Target :
CentOS 8.3.2011
5.10.0-60.18.0.50.aarch64
The NVMe-oF Initiator:
CentOS 8.4.2105
5.15.41
spdk: 22.01
NIC: Mellanox MT27800 CX-5

I have tried to use nvme_perf before and i did not encounter any problem.

**Comment by liuqinfei** [comment]
Here is my result.

[root@client nvme]# startTime_s=`date +%s`;nvme connect -t rdma -a 172.19.23.55 -n nqn.2016-06.io.spdk:cnode1 -s 6001;endTime_s=`date +%s`;sumTime=$[ $endTime_s - $startTime_s ];echo "$startTime ---> $endTime" "Total:$sumTime seconds"
 --->  Total:5 seconds
[root@client nvme]# startTime_s=`date +%s`;nvme disconnect -d /dev/nvme1;endTime_s=`date +%s`;sumTime=$[ $endTime_s - $startTime_s ];echo "$startTime ---> $endTime" "Total:$sumTime seconds"                    --->  Total:10 seconds

**Comment by ksztyber** [comment]
[Bug scrub] Seems like a third-party issue. Aleksey will take a look to confirm this.
